src/main/java/ai/elimu/rest/v2/crowdsource/WordContributionRestController.java

```diff
@@ -0,0 +1,189 @@
+package ai.elimu.rest.v2.crowdsource;
+
+import ai.elimu.dao.*;
+import ai.elimu.model.content.Allophone;
+import ai.elimu.model.content.LetterToAllophoneMapping;
+import ai.elimu.model.content.Word;
+import ai.elimu.model.contributor.Contributor;
+import ai.elimu.model.contributor.WordContributionEvent;
+import ai.elimu.model.v2.gson.content.AllophoneGson;
+import ai.elimu.model.v2.gson.content.LetterToAllophoneMappingGson;
+import ai.elimu.model.v2.gson.content.WordGson;
+import ai.elimu.rest.v2.JpaToGsonConverter;
+import com.google.gson.Gson;
+import org.apache.commons.lang.StringUtils;
+import org.apache.logging.log4j.LogManager;
+import org.apache.logging.log4j.Logger;
+import org.json.JSONArray;
+import org.json.JSONObject;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.http.HttpStatus;
+import org.springframework.http.MediaType;
+import org.springframework.web.bind.annotation.RequestBody;
+import org.springframework.web.bind.annotation.RequestMapping;
+import org.springframework.web.bind.annotation.RequestMethod;
+import org.springframework.web.bind.annotation.RestController;
+
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+import java.util.ArrayList;
+import java.util.Calendar;
+import java.util.List;
+
+/**
+ * REST API for the Crowdsource application: https://github.com/elimu-ai/crowdsource
+ * <p>
+ * This controller is responsible for handling the creation words and fetching data required for the creation of
+ * a new word eg:(Allophones, existing words etc.).
+ */
+
+@RestController
+@RequestMapping(value = "/rest/v2/crowdsource/word-contributions", produces = MediaType.APPLICATION_JSON_UTF8_VALUE)
+public class WordContributionRestController {
+
+    private final Logger logger = LogManager.getLogger();
+
+    @Autowired
+    private WordDao wordDao;
+
+    @Autowired
+    private WordContributionEventDao wordContributionEventDao;
+
+    @Autowired
+    private ContributorDao contributorDao;
+
+    @Autowired
+    private AllophoneDao allophoneDao;
+
+    @Autowired
+    private LetterToAllophoneMappingDao letterToAllophoneMappingDao;
+
+    /**
+     * Handles the creation of a new word & the corresponding word contribution event.
+     *
+     * @param request
+     * @param response
+     * @param requestBody JSON String
```

Comment:
Sure! will add more info about the JSON string, The thing is the requestBody string should contain fields for two types, 
1. fields required for the actual Word Object creation.
2. fields required for the WordContributionEvent creation. 

Is it Ok that the requestBody JSON String contains fields for both types?

Reply:
> Sure! will add more info about the JSON string, The thing is the requestBody string should contain fields for two types,
> 
> 1. fields required for the actual Word Object creation.
> 2. fields required for the WordContributionEvent creation.
> 
> Is it Ok that the requestBody JSON String contains fields for both types?

@UmenR It's a good question. Could you take a look at the current solution for the `AudioContributionsRestController`?: https://github.com/elimu-ai/crowdsource/blob/master/app/src/main/java/ai/elimu/crowdsource/rest/AudioContributionsService.java#L33

There, you'll see that the `providerIdGoogle` is passed as a Header value rather than part of the Body. And there is a second Header value as well, which I realize will probably start to get messy if we in the future add more information about the contribution event. So perhaps passing two types of @RequestBody instances would be better when creating Words (one for the Word itself, and another for the Word contribution event)? Do you know i this is technically possible with the Spring Framework?

Reply:
@nya-elimuai It's currently working as it is, The following would convert the JSON string to a WordGson Object regardless of whether or not the request body contains additional fields that are not defined within `WordGson` Class.
```
WordGson wordGson = new Gson().fromJson(requestBody, WordGson.class);

```
And I do the following to convert the String into a JSONObject instance and then retrieve fields from that instance to instantiate a new WordContributionEvent instance. Note that I use the same `requestBody` in both instances.

```
JSONObject requestJson = new JSONObject(requestBody);
```

However, a more elegant way to handle this would be to create a separate Wrapper Class[[1]](https://stackoverflow.com/questions/5726583/spring-rest-multiple-requestbody-parameters-possible)[[2]](https://stackoverflow.com/questions/27189730/multiple-requestbody-values-in-one-controller-method)[[3]](https://stackoverflow.com/questions/51231070/is-it-possible-to-pass-multiple-objects-in-requestbody/51231222) and define the fields we expect from the request body within it and parse the JSON string creating an instance of the Wrapper Class like so,

```
WrapperClass WrapperClass = new Gson().fromJson(requestBody, WrapperClass.class);
```

And then retrieve the wrapped objects/fields separately. What do you think would be the best approach? 

Leave it as it is, Or define a new `WrapperClass`. If we are to go with the latter we would also have to think of where to place this WrapperClass in our sources. One way is to maybe have it as an inner class of  `WordContributionRestController` so that `WordContributionRestController` itself is responsible for defining what sort of request body it expects in its handler methods.

[1]https://stackoverflow.com/questions/5726583/spring-rest-multiple-requestbody-parameters-possible
[2]https://stackoverflow.com/questions/27189730/multiple-requestbody-values-in-one-controller-method
[3]https://stackoverflow.com/questions/51231070/is-it-possible-to-pass-multiple-objects-in-requestbody/51231222

Reply:
@UmenR I see. Thanks for clarifying.

If we go for the solution of passing two types of Gson objects within the same `requestBody` string, I'm wondering how to best configure this in the Android app. We are currently using Retrofit, so perhaps using something like `@MultiPart` would work?: https://square.github.io/retrofit/

Reply:
@nya-elimuai I missed the whole point of having the Model(https://github.com/elimu-ai/model). 

The Crowdsource Application can send a `WordContributionEventGson` (https://github.com/elimu-ai/model/blob/master/src/main/java/ai/elimu/model/v2/gson/crowdsource/WordContributionEventGson.java).

And the REST API can then use the `WordContributionEventGson` sent in the request body to create a `Word`(https://github.com/elimu-ai/webapp/blob/master/src/main/java/ai/elimu/model/content/Word.java) and a `WordContributionEvent`(https://github.com/elimu-ai/webapp/blob/master/src/main/java/ai/elimu/model/contributor/WordContributionEvent.java) accordingly. This means I don't have to change or add any wrapper classes because the models `WordContributionEventGson` already does this for me. Ill re-request a review once I update the PR.

Reply:
@UmenR Ah, hehe, I forgot all about that also 😆

Yes, you're right. With that solution, we only need to pass one type of object (since a `WordGson` is a property of `WordContributionEventGson`). So this solution will be more elegant.

Good catch! 👍